kcidb/test_oo.py

```diff
@@ -0,0 +1,192 @@
+                        "id": "kernelci:kernelci.org:619c7e1a2e53f8feb4f2efae",
+                        "origin": "kernelci",
+                        "checkout_id": "_:kernelci:5acb9c2a7bc836e9e5172"
+                                       "bbcd2311499c5b4e5f1"
+                    },
+                    {
+                        "id": "redhat:redhat.org:619c65f9709de72e90f2efd0",
+                        "origin": "redhat",
+                        "checkout_id": "_:redhat:5acb9c2a7bc836e9619c6"
+                                       "5f9709de72e90f2efd0"
+                    }
+                ],
+                "tests": [
+                    {
+                        "id": "google:google.org:a19di3j5h67f8d9475f26v11",
+                        "build_id": "google:google.org:a1d993c3n4c448b2"
+                                    "j0l1hbf1",
+                        "origin": "google",
+                    },
+                    {
+                        "waived": False,
+                        "start_time": "2021-11-23T03:52:13.660000+00:00",
+                        "path": "baseline.login",
+                        "status": "PASS",
+                        "id": "kernelci:kernelci.org:619c656de1fb4af479f2efaa",
+                        "build_id": "kernelci:kernelci.org:619c64b171"
+                                    "2847eccbf2efac",
+                        "origin": "kernelci"
+                    },
+                    {
+                        "build_id": "redhat:redhat.org:619c65f970"
+                                    "9de72e90f2efd0",
+                        "id": "redhat:redhat.org:b9d8be63bc2abca63165"
+                              "de5fd74f0f6d2f0b0d1c",
+                        "origin": "redhat",
+                        "waived": True,
+                        "status": "DONE"
+                    },
+                ]
+            }
+        )
+        self.client = kcidb.oo.Client(source)
+
+    def test_client_variable(self):
+        """Check instance for client variable"""
+        self.assertIsInstance(self.client, kcidb.oo.Client)
+
+    def test_returned_instance(self):
+        """Check that the correct instance is returned from the OO Client."""
+
+        self.assertEqual(self.query_str(""), {})
+
+        self.check_type_instance(self.query_str(
+            ">test[kernelci:kernelci.org:619c656de1fb4af479f2efaa]#"),
+            "test", kcidb.oo.Test
+        )
+
+        self.check_type_instance(self.query_str(
+            ">build[kernelci:kernelci.org:619c6a30636e8fe042f2efa0]#"),
+            "build", kcidb.oo.Build
+        )
+
+        self.check_type_instance(self.query_str(
+            ">checkout[_:redhat:5acb9c2a7bc836e9619c65f9709de72e90f2efd0]#"),
+            "checkout", kcidb.oo.Checkout
+        )
+
+        self.check_type_instance(self.query_str(
+            '>revision["5acb9c2a7bc836e9e5172bbcd2311499c5b4e5f1", ""]#'),
+            'revision', kcidb.oo.Revision
+        )
```

Comment:
This is close to "One object, for every type, in turn", but doesn't check that only one instance of each object was returned.